### DISCUSSION TITLE
Nightly (maybe even automated deployment version) somewhere publicly?

### DISCUSSION THREAD

**Discussion by rugk** [PrivateBin/PrivateBin#1422] [Ideas]
For debugging and reproducing issues, it would be great if we'd had some nightly version running somewhere publicly with the lastest main branch code running?

E.g. for debugging https://github.com/orgs/PrivateBin/discussions/1405 this would be very much very helpful, I guess hmm…?

Maybe the deployment could just be automated by publishing the container… though ah it does not use the latest commit code, hmm…

Well… just an idea, seems more complicated than I think. And I can use GitHub Codespaces, which also sets it up kinda I just remembered.

**Comment by elrido** [comment]
Creating it using the containers isn't complicated, but hosting it would require frequent updates, which would have to get automated. IMO it is better to simplify the devcontainer process here on github, so folks can more easily spin up their own instance and iterate on the code. I can share my scripts (a one-liner really) that I use to spin up one of our official images, but bind mount my development git repo into it, which lets me instantly test any changes done in my local browser (including seing that I didn't update the SRI hashes and it therefore not loading, etc.).

**Reply by rugk** [reply]
Yeah I mean after all devcontainer kinda is that one-liner, only that you work _inside_ the devcontainer.

But yeah, I also found that solution and IMHO, it's indeed a good solution for the problem.
